code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -65,8 +65,8 @@
 		if(canpass && CANATMOSPASS(current_turf, src, (direction & (UP|DOWN))) && !(blocks_air || current_turf.blocks_air))
 			LAZYINITLIST(atmos_adjacent_turfs)
 			LAZYINITLIST(current_turf.atmos_adjacent_turfs)
-			atmos_adjacent_turfs[current_turf] = TRUE
-			current_turf.atmos_adjacent_turfs[src] = TRUE
+			atmos_adjacent_turfs += current_turf
+			current_turf.atmos_adjacent_turfs += src
```

Comment:
this would double up thou and |= might eat in to your savings, also `in` is slower.

I have a better idea:

Use log2() to reduce a directional into a bit index (turns 1 to 0, 2 into 1, 4 into 2) (adding +1 for 1 indexed lists) and store a list of list[log2(direction)+1] = turf|null

this keeps direct access sped up (the macro at the top), since you can access by direction, loops sped up, and prevents duping.

the most this method could ever reach is 6 turfs, and lists pre-init with 7 items of reserve, so no more memory then only storing adjacent turfs